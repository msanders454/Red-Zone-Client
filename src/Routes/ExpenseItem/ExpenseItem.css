.expense-container {
    width: 80vw;
    border: 1px solid rgb(59, 6, 6);
    border-radius: 15px;
    background-color: white;
    text-align: center;
    margin-bottom: 20px;
    padding: 2%;
    font-weight: 500;
}

.expense-item {
    padding-bottom: 1%;
}

.edit-delete {
    font-size: 25px;
    cursor: pointer;
    color: rgb(255, 0, 0);
}


.buttonDelete {
    background-color: transparent;
    border: 1px;
    cursor: pointer;
    overflow: hidden;
    outline: none;
}

.buttonDelete:hover {
    background-color: red;
    font-size: 15px;
    color: white
}

a {
    color: black;
    text-decoration: none;
  }

  a:hover {
    color: white
}


@media only screen and (min-width: 768px) {
    .expense-container {
        margin: 15px 15%;
        display: flex;
        padding: 0;
    }
    
    .expense-header {
        display: flex;
        justify-content: space-between;
    }
    .expense-item {
        padding: 1.5rem 2rem;
        width: 8vw;
    }
    .ExpenseItem__Info {
        font-size: 0.8rem;
    }
    .edit-delete {
        height: 80px;
        margin-left: 15px;
    }
}

@media only screen and (min-width: 968px) {
    .expense-item {
        width: 9vw;
    }
    .ExpenseItem__Info {
        font-size: 0.9rem;
    }
}

@media only screen and (min-width: 1168px) {
    .expense-container {
        width: 75vw;
    }
    .expense-item {
        width: 10vw;
        padding-top: 3.75%;
    }
    .ExpenseItem__Info {
        font-size: 1rem;
    }
}

@media only screen and (min-width: 1368px) {
    .expense-container {
        width: 70vw;
    }
}