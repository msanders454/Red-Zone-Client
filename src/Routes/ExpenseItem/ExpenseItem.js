import React from 'react';
import RedZoneContext from '../../RedZoneContext';
import config from '../../config';
import { Link } from 'react-router-dom';
import './ExpenseItem.css';

function deleteExpense(expenseId, callback) {
    let confirmDelete = window.confirm('Confirm to delete this expense? Deleted expenses CANNOT be recovered!');
    if (confirmDelete) {
        fetch(config.API_ENDPOINT + `/${expenseId}`, {
            method: "DELETE",
            headers: {
                "content-type": "application/json"
            }
        })
            .then(res => {
                if (!res.ok) return res.json().then(error => Promise.reject(error));
            })
            .then(data => {;
                callback(expenseId);
            })
            .catch(error => {
            });
    }
}

export default function ExpenseItem(props) {
    return (
        <RedZoneContext.Consumer>
            {context => (
                <div className='expense-container'>

                    <div className='expense-header'>
                    <div className='expense-item'>
                        <span className='ExpenseItemInfo'>{props.description}</span>
                        </div>

                        <div className='expense-item'>
                            <span className='ExpenseItemInfo'>{props.date}</span>
                        </div>

                        <div className='expense-item'>
                            <span className='ExpenseItemInfo'>${props.amount}</span>
                        </div>

                        <div className='expense-item'>
                            <span className='ExpenseItemInfo'>{props.style}</span>
                        </div>

                    </div>

                    <button className='buttonDelete'>
                        <Link to={`/update/${props.id}`}>
                            Update
                        </Link>
                    </button>
                    <button
                        onClick={() => {
                            deleteExpense(props.id, context.deleteExpense);
                        }}
                        className='buttonDelete'
                    >
                       Delete 
                    </button>
                </div>
            )}
        </RedZoneContext.Consumer>
    );
}

ExpenseItem.defaultProps = {
    onClickDelete: () => {},
}
