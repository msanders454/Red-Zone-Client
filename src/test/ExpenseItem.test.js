import React from "react";
import { shallow, configure } from "enzyme";
import ExpenseItem from "../Routes/ExpenseItem/ExpenseItem";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
test("should render ExpenseItem correctly", () => {
  const wrapper = shallow(<ExpenseItem />);
  expect(wrapper).toMatchSnapshot();
});