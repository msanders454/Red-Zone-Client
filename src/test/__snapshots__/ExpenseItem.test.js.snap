// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseItem correctly 1`] = `ShallowWrapper {}`;
